homeassistant/components/aosmith/sensor.py

```diff
@@ -0,0 +1,84 @@
+"""The sensor platform for the A. O. Smith integration."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import AOSmithData
+from .const import DOMAIN, HOT_WATER_STATUS_MAP
+from .coordinator import AOSmithCoordinator
+from .entity import AOSmithEntity
+
+
+@dataclass
+class AOSmithRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[dict[str, Any]], str | int | None]
+
+
+@dataclass
+class AOSmithSensorEntityDescription(SensorEntityDescription, AOSmithRequiredKeysMixin):
+    """Define sensor entity description class."""
+
+
+ENTITY_DESCRIPTIONS: tuple[AOSmithSensorEntityDescription, ...] = (
+    AOSmithSensorEntityDescription(
+        key="hot_water_availability",
+        translation_key="hot_water_availability",
+        icon="mdi:water-thermometer",
+        device_class=SensorDeviceClass.ENUM,
+        options=["low", "medium", "high"],
+        value_fn=lambda device: HOT_WATER_STATUS_MAP.get(
+            device.get("data", {}).get("hotWaterStatus")
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up A. O. Smith sensor platform."""
+    data: AOSmithData = hass.data[DOMAIN][entry.entry_id]
+
+    entities = []
+
+    for description in ENTITY_DESCRIPTIONS:
+        for junction_id in data.coordinator.data:
+            entities.append(
+                AOSmithSensorEntity(data.coordinator, description, junction_id)
+            )
+
+    async_add_entities(entities)
```

Comment:
Can we make this 

Async add entities(
Sensor
For description in descriptions
For data in coordinator
)

I don't know the right name for this, but you probably know what I mean